### PR TITLE
Check httpRes is not nil and select subregion for subnet

### LINKED ISSUE #199: Check httpRes is not nil
### Explain problem to solve

When we call outscale api, httpres can be nil and httpres.Status return nil pointer exception

### Describe the solution you would like

Check httpRes is not nil.

### Additional context

None

### Environment

```markdown
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version:
- cluster-api version:
- Install tools:
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

### LINKED ISSUE #203: Choose subregionName for subnet
### What happened

If we want to use the second az (us-east-2b, eu-west-2b, ...), we can not because subregion will always use the first az.

### Step to reproduce

Create a cluster with us-east-2b, eu-west-2b.

### Expected to happen

We can create a cluster on the second az

### Add anything

None

### cluster-api output

None

### Environment

```markdown
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version:
- cluster-api version: 
- Install tools:
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

### PR DESCRIPTION
**Pull request by outscale-vbr**
**What type of PR is this?**
/kind bug 

**What this PR does / why we need it**:
Check httpRes is not nil, return error otherwise.
httpRes can be nil and httpRes.status return nil pointer exception.
We can also select subregion for subnet

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #199 
Fixes #203 
**Special notes for your reviewer**:
We also select subregion
_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Comment by outscale-vbr** [comment]
@outscale-mdr @outscale-hmi   I supposed that every api call can not return http status.

**Comment by outscale-vbr** [comment]
Closes #199

**Comment by outscale-vbr** [comment]
@outscale-mdr @outscale-hmi Are you ok with the latest version ?

File: cloud/services/compute/image.go [hunk 1]
```diff
@@ -38,8 +38,12 @@ func (s *Service) GetImage(imageId string) (*osc.Image, error) {
 	oscAuthClient := s.scope.GetAuth()
 	readImageResponse, httpRes, err := oscApiClient.ImageApi.ReadImages(oscAuthClient).ReadImagesRequest(readImageRequest).Execute()
 	if err != nil {
-		fmt.Printf("Error with http result %s", httpRes.Status)
-		return nil, err
+		if httpRes != nil {
+			fmt.Printf("Error with http result %s", httpRes.Status)
+			return nil, err
```
Comment (outscale-mdr) [inline cloud/services/compute/image.go:43]: You shoud return the error and not Printf

```fmt.Errorf("error %s httpRes %s", err, httpRes.Status)```

File: cloud/services/compute/image.go [hunk 2]
```diff
@@ -38,8 +38,12 @@ func (s *Service) GetImage(imageId string) (*osc.Image, error) {
...
+		} else {
+			return nil, fmt.Errorf("%w Can not getImage with empty httpRes", err)
+		}
```
Comment (outscale-mdr) [inline cloud/services/compute/image.go:46]: No need to specify "with empty Res", just return the error

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

**Comment by outscale-vbr** [comment]
@outscale-mdr Are you ok with latest changes

**Review by outscale-mdr — APPROVED** [review]
Do not forget the doc :)

**Comment by outscale-vbr** [comment]
@outscale-mdr Thanks i add the docs.

**Pull request merged**